src/lib.rs

```diff
@@ -1,62 +1,196 @@
+                return Ok(JsValue::from(array));
+            }
+        } else {
+            unsafe {
+                let array = js_sys::Uint8Array::view(&mut result.serialize_uncompressed());
+                return Ok(JsValue::from(array));
+            }
+        }
+    }
+    #[wasm_bindgen(js_name = pointAddScalar)]
+    #[allow(unused_variables)]
+    pub fn point_add_scalar(
+        &self,
+        p: JsBuffer,
+        tweak: JsBuffer,
+        compressed: Option<bool>,
+    ) -> Option<JsBuffer> {
+        Some(Box::new([0u8]))
+    }
+    #[wasm_bindgen(js_name = pointCompress)]
+    #[allow(unused_variables)]
+    pub fn point_compress(&self, p: JsBuffer, compressed: Option<bool>) -> JsBuffer {
+        Box::new([0u8])
+    }
+    #[wasm_bindgen(js_name = pointFromScalar)]
+    #[allow(unused_variables)]
+    pub fn point_from_scalar(&self, d: JsBuffer, compressed: Option<bool>) -> Option<JsBuffer> {
+        Some(Box::new([0u8]))
+    }
+    #[wasm_bindgen(js_name = pointMultiply)]
+    #[allow(unused_variables)]
+    pub fn point_multiply(
+        &self,
+        p: JsBuffer,
+        tweak: JsBuffer,
+        compressed: Option<bool>,
+    ) -> Option<JsBuffer> {
+        Some(Box::new([0u8]))
+    }
+    #[wasm_bindgen(js_name = privateAdd)]
+    #[allow(unused_variables)]
+    pub fn private_add(&self, d: JsBuffer, tweak: JsBuffer) -> Option<JsBuffer> {
+        Some(Box::new([0u8]))
+    }
+    #[wasm_bindgen(js_name = privateSub)]
+    #[allow(unused_variables)]
+    pub fn private_sub(&self, d: JsBuffer, tweak: JsBuffer) -> Option<JsBuffer> {
+        Some(Box::new([0u8]))
+    }
+    #[wasm_bindgen]
+    pub fn sign(&self, hash: JsBuffer, x: JsBuffer) -> Result<JsBuffer, JsValue> {
+        set_panic_hook();
+        // How do I check Buffer.isBuffer()?
+        if hash.len() != 32 {
+            return Err(JsValue::from(TypeError::new("Expected Hash")));
+        }
+        if x.len() != 32 {
+            return Err(JsValue::from(TypeError::new("Expected Private")));
+        }
+        let msg_hash = Message::from_slice(&hash).unwrap_throw();
+        let pk = SecretKey::from_slice(&x).unwrap_throw();
+        Ok(Box::new(self.secp.sign(&msg_hash, &pk).serialize_compact()))
+    }
+    #[wasm_bindgen(js_name = signWithEntropy)]
+    #[allow(unused_variables)]
+    pub fn sign_with_entropy(&self, hash: JsBuffer, x: JsBuffer, add_data: JsValue) -> JsBuffer {
+        Box::new([0u8])
+    }
+    #[wasm_bindgen]
+    #[allow(unused_variables)]
+    pub fn verify(
+        &self,
+        hash: JsBuffer,
+        q: JsBuffer,
+        signature: JsValue,
+        compressed: Option<bool>,
+    ) -> bool {
+        // strict flag is not found in js impl (See #5)
+        true
+    }
 }
-#[wasm_bindgen]
-pub fn verify(hash: JsBuffer,  q: JsBuffer, signature: JsValue, compressed: Option<bool>) -> bool { // strict flag is not found in js impl (See #5)
-    true
+
+fn set_panic_hook() {
```

Comment:
Call it only once while initializing
https://rustwasm.github.io/book/reference/debugging.html#logging-panics
You can call initialization by `#[wasm_bindgen(start)]`
https://rustwasm.github.io/docs/wasm-bindgen/reference/attributes/on-rust-exports/start.html